package/Dockerfile

```
@@ -1,3 +1,7 @@
-FROM alpine
+FROM registry.suse.com/suse/sle15:15.3
+RUN zypper -n update && \
+    zypper -n clean -a && \
+    rm -rf /tmp/* /var/tmp/* /usr/share/doc/packages/* /usr/share/doc/manual/* /var/log/*
```

Since you really don't need any additional packages, why not use `bci-micro` instead? It's much smaller
```suggestion
FROM registry.suse.com/bci/bci-micro:15.3
```
Note, that `bci-micro` doesn't ship zypper, so the `RUN` part has to be dropped.

---

@macedogm Any comments on the above? I think that the final image still uses bci-base (rather than micro/minimal) - would you mind elaborating on why?

---

I understand doing that in the case of the core rancher containers, but does this really provide value in the case of side functionality like the webhook? I would also expect that running as a non-root, non-wheeled user in the container would prevent you from using zypper in the first place - if you disagree let me know.

---

You are right in the rationale. When running as non-root user we could move directly to [bci/bci-minimal](https://registry.suse.com/static/bci/bci-minimal/index.html) or [bci/bci-micro](https://registry.suse.com/static/bci/bci-micro/index.html), but for the moment they are still in `Tech Preview`. I would say that we could stay with `bci/bci-base` for now and then switch.

---

Another point is that `bci/bci-minimal` and `bci/bci-micro` don't have `useradd` command, so either we create the user by hand, ask BCI team to add it or stick with the `USER` command in Dockerfile. Either way we have some trade-offs to choose.

---

If the busybox userland is sufficient for you, then you could also use `bci/busybox`, which ships `useradd` (the busybox implementation though so there might be :dragon:).

---

> I did not realize that minimal was in tech preview - good point (though micro seems to be out of tech preview for the non-latest tags). 

All of the images, except `bci-base` should be tech-preview. Note that this really just means that the image is not officially supported by SUSE's commercial support and that's about it.

> As far as `useradd` goes - that's a really good point that we should mention regardless of what we end up doing here. @dcermak do you happen to know why these aren't included in these images? The "add a non-root user and use that" pattern is fairly common (if I am not mistaken).

We haven't included it, because no one requested it and we only put the bare minimum in. You could work around this with the minimal image though by downloading the useradd rpm in a previous stage, copying it into the minimal image, installing it via rpm, adding the user and then removing it in the same step. Or you can make a request for useradd to be included in these images.